src/config/agent_config.go

```diff
@@ -30,6 +31,7 @@ func ProcessArgs() {
 	flag.StringVar(&ret.TargetHost, "host", client.DefaultHost, "The target host")
 	flag.IntVar(&ret.TargetPort, "port", 80, "The target port")
 	flag.StringVar(&ret.ClusterID, "cluster-id", "default-cluster", "The value of the cluster-id")
+	flag.StringVar(&ret.AgentVersion, "agent-version", "", "Discovery agent version")
```

Comment:
Shouldn't the version be hard-coded? Something that cannot change by parameter?
Maybe it should be a generated file - can be invoked from makefile or "go generate" and be based on git version/tag

Reply:
if it can be done automatically when we push then yes. if i need to run `go generate` then none will really going to do it.

Reply:
Assuming that it is the correct way to provide version, then seems that the problem is solvable.  Isn't it?

Reply:
currently i'm not working with hashes for all the other version fields, so making it different will be strange UX.
currently all the versions are the image name.

Reply:
OK